scripts/fregion.py

```diff
@@ -1015,41 +1122,44 @@ def meta(f): return FRMeta(f)
 #
 #######
 
+def RegReader(desc):
+  def newCls(cls):
+    FRegion.addType(desc, lambda renv, ty, repty: cls(renv, ty, repty))
+    return cls
+  return newCls
+
 # date/time
+@RegReader("datetime")
```

Comment:
This is great, simpler to read as a self-contained definition.

Reply:
Yes. that is python. I really hope c++ can add this feature in future